android/src/main/java/com/reactnativecommunity/cameraroll/CameraRollModule.java

```diff
@@ -514,15 +520,20 @@ private static boolean putImageInfo(
   private static void putLocationInfo(
       Cursor media,
       WritableMap node,
-      int longitudeIndex,
-      int latitudeIndex) {
-    double longitude = media.getDouble(longitudeIndex);
-    double latitude = media.getDouble(latitudeIndex);
-    if (longitude > 0 || latitude > 0) {
-      WritableMap location = new WritableNativeMap();
-      location.putDouble("longitude", longitude);
-      location.putDouble("latitude", latitude);
-      node.putMap("location", location);
-    }
+      int dataIndex,
+      ExifInterface exif) {
+      // location details are no longer indexed for privacy reasons using string Media.LATITUDE, Media.LONGITUDE
+      // we manually obtain location metadata using ExifInterface#getLatLong(float[]).
+      // ExifInterface is added in API level 5
+      float[] imageCoordinates = new float[2];
+      boolean hasCoordinates = exif.getLatLong(imageCoordinates);
+      if (hasCoordinates) {
+        double longitude = imageCoordinates[1];
+        double latitude = imageCoordinates[0];
+        WritableMap location = new WritableNativeMap();
+        location.putDouble("longitude", longitude);
+        location.putDouble("latitude", latitude);
+        node.putMap("location", location);
+      }
```

Comment:
Is there a reason why you ditched the try-catch around this code block? I think an error in this code block would cause the whole media query to stop and reject the promise to javascript. With all Android privacy and scoped storage changes going on we can't fully rely on happy execution or parsing. I think we should redo the try-catch so that it can not break the whole query. What do you think?

```suggestion
      try {
            // location details are no longer indexed for privacy reasons using string Media.LATITUDE, Media.LONGITUDE
            // we manually obtain location metadata using ExifInterface#getLatLong(float[]).
            // ExifInterface is added in API level 5
            float[] imageCoordinates = new float[2];
            boolean hasCoordinates = exif.getLatLong(imageCoordinates);
            if (hasCoordinates) {
                  double longitude = imageCoordinates[1];
                  double latitude = imageCoordinates[0];
                  WritableMap location = new WritableNativeMap();
                  location.putDouble("longitude", longitude);
                  location.putDouble("latitude", latitude);
                  node.putMap("location", location);
            }
      } catch (IOException e) {
            FLog.e(ReactConstants.TAG, "Could not read the metadata", e);
      }
```

Reply:
The try catch is moved to getExifInterface

Reply:
Catching an `IOException` at this point doesn't make much sense indeed. Catching a more generic Exception is also not really needed.